### PR TITLE
StandUps

### LINKED ISSUE #341: Daily ping for Standup
Every day at 8 AM (?), the bot will ping the @ Standup role.

The second phase could be that the bot will analyze all yesterday's messages and make stats about how many ✅, ❌ and in-progress results are there.

### PR DESCRIPTION
**Pull request by tenhobi**
Fixes #341

### CODE REVIEW

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    /// [] - in progress during the day, failed later
+    /// [:white_check_mark:] - completed
+    ///
+    /// priority:
+    /// [] - normal
+    /// []! - critical
+    /// </summary>
+    private static readonly Regex s_regex = new(@"^\s*\[\s*(?<State>\S*)\s*\]\s*(?<Priority>[!])?");
+
+    private static readonly List<string> s_okList = new() { "check", "done", "ok", "✅" };
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs]:
```suggestion
    private static readonly Regex Regex = new(@"^\s*\[\s*(?<State>\S*)\s*\]\s*(?<Priority>[!])?");

    private static readonly List<string> OkList = new() { "check", "done", "ok", "✅" };
```

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    public override string ToString()
+    {
+        return $"{Sum} ({_normal} + {_must}!)";
+    }
+
+    public StandUpStats Add(StandUpStats other)
+    {
+        return new StandUpStats { _normal = _normal + other._normal, _must = _must + other._must };
+    }
+}
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs]: Separate file

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+    private static readonly Regex s_regex = new(@"^\s*\[\s*(?<State>\S*)\s*\]\s*(?<Priority>[!])?");
+
+    private static readonly List<string> s_okList = new() { "check", "done", "ok", "✅" };
+
+    public string Name { get; } = "standup";
+
+    public async Task ExecuteAsync(CancellationToken cancellationToken)
+    {
+        var today = DateTime.Today; // Fix one point in time.
+        await SendStandUpNotification(today);
```
Comment (ostorc) [inline src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs]: Why did you put it into separate method?

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+                int messagesCount = messageList.Count;
+                messageList.AddRange(
+                    await channel.GetMessagesBeforeAsync(messageList.Last().Id)
+                );
+
+                // No new data.
+                if (messageList.Count <= messagesCount)
+                {
+                    break;
+                }
```
Comment (stepech) [inline src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs:79]: Big no

**Review by stepech — CHANGES_REQUESTED** [review]

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs (hunk 4 above)
Comment (tenhobi) [reply src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs:79]: Big yes, you are dumb.

**Review by stepech — APPROVED** [review]

File: src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs (hunk 4 above)
Comment (tenhobi) [reply src/HonzaBotner.Discord.Services/Jobs/StandUpJobProvider.cs:79]: Big yes, you are dumb.

**Pull request merged**